From 6bcce5ea56614dbcfdc37e7dcf690ea660d96b78 Mon Sep 17 00:00:00 2001
From: Aran Dalton <arda@allwinnertech.com>
Date: Tue, 24 Mar 2026 15:31:32 +0800
Subject: [PATCH 44/63] ANDROID: GKI: update symbol list for allwinner

2 function symbol(s) added
  'loff_t no_seek_end_llseek(struct file*, loff_t, int)'
  'void skb_append(struct sk_buff*, struct sk_buff*, struct sk_buff_head*)'

Bug: 495655551
Bug: 498402639
Change-Id: I42b6ebde21b6f8edaae8cabb30955084a59ab31c
Signed-off-by: Aran Dalton <arda@allwinnertech.com>
(cherry picked from commit ff78c78d65d3b2e46c6cd4c1b59e8e0e6150cdae)
---
 gki/aarch64/abi.stg           | 27 +++++++++++++++++++++++++++
 gki/aarch64/symbols/allwinner |  2 ++
 2 files changed, 29 insertions(+)

diff --git a/gki/aarch64/abi.stg b/gki/aarch64/abi.stg
index 7ef34e7607c8..90d68d4d6d07 100644
--- a/gki/aarch64/abi.stg
+++ b/gki/aarch64/abi.stg
@@ -379346,6 +379346,13 @@ function {
   return_type_id: 0x48b5725f
   parameter_id: 0x05003835
 }
+function {
+  id: 0x11d96a86
+  return_type_id: 0x48b5725f
+  parameter_id: 0x054f691a
+  parameter_id: 0x054f691a
+  parameter_id: 0x1f73dead
+}
 function {
   id: 0x11ddd684
   return_type_id: 0x48b5725f
@@ -492828,6 +492835,15 @@ elf_symbol {
   type_id: 0x15334a4f
   full_name: "nla_strscpy"
 }
+elf_symbol {
+  id: 0x6e09b6e4
+  name: "no_seek_end_llseek"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x72c16011
+  type_id: 0x2d64ae3e
+  full_name: "no_seek_end_llseek"
+}
 elf_symbol {
   id: 0x39787440
   name: "node_states"
@@ -508134,6 +508150,15 @@ elf_symbol {
   type_id: 0x104ab2ae
   full_name: "skb_add_rx_frag_netmem"
 }
+elf_symbol {
+  id: 0x34556105
+  name: "skb_append"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0xf04a2e6a
+  type_id: 0x11d96a86
+  full_name: "skb_append"
+}
 elf_symbol {
   id: 0x736f4bcb
   name: "skb_append_pagefrags"
@@ -532454,6 +532479,7 @@ interface {
   symbol_id: 0x49027247
   symbol_id: 0x85e07b67
   symbol_id: 0x6359420c
+  symbol_id: 0x6e09b6e4
   symbol_id: 0x39787440
   symbol_id: 0x7f258c4e
   symbol_id: 0x754435c8
@@ -534153,6 +534179,7 @@ interface {
   symbol_id: 0xc4c0ed9a
   symbol_id: 0x37d54444
   symbol_id: 0x5516a755
+  symbol_id: 0x34556105
   symbol_id: 0x736f4bcb
   symbol_id: 0x3ffd8828
   symbol_id: 0x38f9f2a7
diff --git a/gki/aarch64/symbols/allwinner b/gki/aarch64/symbols/allwinner
index 64508d537c23..dd7dc449b3e7 100644
--- a/gki/aarch64/symbols/allwinner
+++ b/gki/aarch64/symbols/allwinner
@@ -87,3 +87,5 @@
   clk_unregister_fixed_rate
   clk_unregister_gate
   pinctrl_gpio_set_config
+  skb_append
+  no_seek_end_llseek
-- 
2.50.1

